.github/scripts/check-pr-has-required-labels.ts

```diff
@@ -0,0 +1,97 @@
+import * as core from '@actions/core';
+import { context, getOctokit } from '@actions/github';
+import { GitHub } from '@actions/github/lib/utils';
+
+main().catch((error: Error): void => {
+  console.error(error);
+  process.exit(1);
+});
+
+async function main(): Promise<void> {
+  // "GITHUB_TOKEN" is an automatically generated, repository-specific access token provided by GitHub Actions.
+  const githubToken = process.env.GITHUB_TOKEN;
+  if (!githubToken) {
+    core.setFailed('GITHUB_TOKEN not found');
+    process.exit(1);
+  }
+
+  // Initialise octokit, required to call Github GraphQL API
+  const octokit: InstanceType<typeof GitHub> = getOctokit(githubToken);
+
+  // Retrieve pull request info from context
+  const prRepoOwner = context.repo.owner;
+  const prRepoName = context.repo.repo;
+  const prNumber = context.payload.pull_request?.number;
+  if (!prNumber) {
+    core.setFailed('Pull request number not found');
+    process.exit(1);
+  }
+
+  // Retrieve pull request labels
+  const prLabels = await retrievePullRequestLabels(octokit, prRepoOwner, prRepoName, prNumber);
+
+  const preventMergeLabels = ["needs-qa", "QA'd but questions", "issues-found"];
```

Comment:
we can also add labels such as `need-ux-ds-review` (which i see a lot of this one), `blocked` (provided but maybe not used as many` , `stale`, `DO-NOT-MERGE`, for now, wdyt?

Reply:
There is already [a CI workflow](https://github.com/MetaMask/metamask-extension/blob/develop/.github/workflows/do-not-merge.yml) to prevent merging `DO-NOT-MERGE` labelled PRs. Though we may want to consolidate it here, that's a good point.

Though... we could move in the other direction as well. The DO-NOT-MERGE workflow is a lot simpler. We can access the necessary information (PR labels) purely using GitHub Actions context, no queries needed.

Reply:
Thanks for your feedbacks @DDDDDanica and @Gudahtt . We've discussed this with @danjm @sethkfman @cortisiko @PeterYinusa @vpintorico last Friday and agreed on:
- add checks for these labels in this new Github action: `need-ux-ds-review`, `blocked`, `stale`, `DO-NOT-MERGE`
- remove the old [DO-NOT-MERGE Github action](https://github.com/MetaMask/metamask-extension/blob/develop/.github/workflows/do-not-merge.yml) as duplicate of this new Github action